### PR TITLE
Promote ApiMayChange features #26532

### PR DESCRIPTION
**Pull request by helena**
## Purpose
Promote ApiMayChange features for 2.6. More will be promoted as we get closer which are not included here. These are ready.

## References
https://github.com/akka/akka/issues/26532 clarifies what we are not promoting until we are closer to the release.

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

File: akka-stream/src/main/scala/akka/stream/javadsl/StreamRefs.scala [hunk 1]
```diff
@@ -28,7 +26,6 @@ object StreamRefs {
    *
    * See more detailed documentation on [[SourceRef]].
    */
-  @ApiMayChange
   def sourceRef[T](): javadsl.Sink[T, CompletionStage[SourceRef[T]]] =
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/javadsl/StreamRefs.scala]: Let's try to do https://github.com/akka/akka/issues/24372 before removing the ApiMayChange for StreamRefs. That should be a separate PR, so just revert in this file here.

File: akka-stream/src/main/scala/akka/stream/scaladsl/StreamRefs.scala [hunk 2]
```diff
@@ -29,7 +27,6 @@ object StreamRefs {
    *
    * See more detailed documentation on [[SourceRef]].
    */
-  @ApiMayChange
   def sourceRef[T](): Sink[T, Future[SourceRef[T]]] =
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/scaladsl/StreamRefs.scala]: revert in this file until we have tried to fix #24372

**Review by patriknw — COMMENTED** [review]
good, but let's try to fix one thing in StreamRefs before promoting that

**Comment by patriknw** [comment]
also search for `API MAY CHANGE` as Scaladoc comment

File: akka-stream/src/main/scala/akka/stream/javadsl/StreamRefs.scala (hunk 1 above)
Comment (patriknw) [reply akka-stream/src/main/scala/akka/stream/javadsl/StreamRefs.scala]: I have fixed StreamRefs in https://github.com/akka/akka/pull/26847

Comment (helena) [reply akka-stream/src/main/scala/akka/stream/javadsl/StreamRefs.scala]: OK, I will update docs.

File: akka-stream/src/main/scala/akka/stream/javadsl/FlowWithContext.scala [hunk 3]
```diff
@@ -19,7 +18,6 @@ import scala.compat.java8.FutureConverters._
 /**
  * API MAY CHANGE
  */
-@ApiMayChange
```
Comment (raboof) [inline akka-stream/src/main/scala/akka/stream/javadsl/FlowWithContext.scala]: Hmm, this API is fairly new as well - even though we didn't mention it in #26532 I wonder if we should already remove the ApiMayChange here?

If we decide this API is ready to be set into stone, we should also remove the `API MAY CHANGE` scaladoc here

**Review by raboof — COMMENTED** [review]
Looks good. Are we really ready to declare the FlowWithContext API's stable?

File: akka-stream/src/main/scala/akka/stream/javadsl/FlowWithContext.scala (hunk 3 above)
Comment (helena) [reply akka-stream/src/main/scala/akka/stream/javadsl/FlowWithContext.scala]: @raboof If you are unsure then I'd suggest we keep it there until we are. I'm working on the scaladoc removals where relevant 👍

**Comment by helena** [comment]
Not finding any usage of it in scaladoc where the annotation on code is now removed.

**Comment by helena** [comment]
Updated now to remove API MAY CHANGE from relevant scaladoc and reverted FlowWithContext.

File: akka-stream/src/main/scala/akka/stream/javadsl/FlowWithContext.scala (hunk 3 above)
Comment (patriknw) [reply akka-stream/src/main/scala/akka/stream/javadsl/FlowWithContext.scala]: ok, we can do another sweep closer to the final
(I don't think we guarantee anything for milestones, but better to avoid confusion in case we still need to change those)

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala [hunk 4]
```diff
@@ -205,7 +203,6 @@ abstract class EventSourcedBehavior[Command, Event, State] private[akka] (
  * There will be compilation errors if the returned effect isn't a [[ReplyEffect]], which can be
  * created with `Effects().reply`, `Effects().noReply`, [[Effect.thenReply]], or [[Effect.thenNoReply]].
  */
-@ApiMayChange
```
Comment (patriknw) [inline akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala]: this is in Typed

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala [hunk 5]
```diff
@@ -3401,10 +3400,7 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
    *
    * @param collapseContext turn each incoming pair of element and context value into an element of this Flow
    * @param extractContext turn each outgoing element of this Flow into an outgoing context value
-   *
-   * API MAY CHANGE
    */
-  @ApiMayChange
   def asFlowWithContext[U, CtxU, CtxOut](
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: WithContext should wait?

File: akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala (hunk 4 above)
Comment (helena) [reply akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/EventSourcedBehavior.scala]: Ouch, fixed thanks.

File: akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala (hunk 5 above)
Comment (helena) [reply akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala]: Yes, fixed.

File: akka-stream/src/main/scala/akka/stream/javadsl/Source.scala [hunk 6]
```diff
@@ -3570,10 +3569,6 @@ final class Source[Out, Mat](delegate: scaladsl.Source[Out, Mat]) extends Graph[
   def log(name: String): javadsl.Source[Out, Mat] =
     this.log(name, ConstantFun.javaIdentityFunction[Out], null)
 
-  /**
-   * API MAY CHANGE
-   */
-  @ApiMayChange
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/javadsl/Source.scala]: WithContext is in several places

Comment (helena) [reply akka-stream/src/main/scala/akka/stream/javadsl/Source.scala]: @patriknw ping, I think I pushed the last of WithContext now.

**Review by patriknw — APPROVED** [review]
LGTM
